lib/msf/base/sessions/command_shell.rb

```diff
@@ -421,52 +421,72 @@ def cmd_download(*args)
 
     # Check if src exists
     if !_file_transfer.file_exist?(src)
-      print_error("The target file does not exist")
+      print_error('The target file does not exist')
       return
     end
 
+    fs_sep = platform == 'windows' ? '\\' : '/'
+    if dst.blank?
+      dst = src.split(fs_sep).last
```

Comment:
The only problem I forsee with this is that we don't check if `src` itself is a directory prior to doing this split, which means there is a possibility that `dst` is incorrectly set.

Reply:
Source directories are not supported. About 7 lines prior to this, it's validated that `src` is in fact a file that exists.